packages/eventindexer/indexer/save_block_verified_event.go

```diff
@@ -0,0 +1,91 @@
+package indexer
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/taikoxyz/taiko-mono/packages/eventindexer"
+	"github.com/taikoxyz/taiko-mono/packages/eventindexer/contracts/taikol1"
+)
+
+func (svc *Service) saveBlockVerifiedEvents(
+	ctx context.Context,
+	chainID *big.Int,
+	events *taikol1.TaikoL1BlockVerifiedIterator,
+) error {
+	if !events.Next() || events.Event == nil {
+		log.Infof("no BlockVerified events")
+		return nil
+	}
+
+	for {
+		event := events.Event
+
+		if event.Raw.Removed {
+			continue
```

Comment:
Do we need to remove the corresponding saved event data in database? or simply ignoring it is fine?

Reply:
Yeah, none of the event handling removes from DB yet (same with block proposed and block proven still), no reorg handling. I can do it for all 3 in this PR.

Reply:
actually, I think i will do a separate PR to handle reorg entirely, similiar to the taiko-client one, since it will make this PR quite large and take some time.